app/controllers/api/submissions_controller.rb

```diff
@@ -9,10 +9,14 @@ def create
 
       return render_solution_not_accessible unless solution.user_id == current_user.id
 
+      formatted_files = submission_params[:files].each_with_object({}) do |file, files|
+        files[file[:name]] = file[:contents]
+      end
```

Comment:
I'm happy to change this, but would rather we change it so that `PrepareMappedFiles` expects an array, not a hash. ie, move this code into `Submission::PrepareMappedFiles`, and update the tests accordingly.

The main purpose of `Submission::PrepareMappedFiles` is to do this array/hash manipulation in one single place, so I don't want to preprocess in the controller before we get there.